utility/tests/test_repoutil.py

```diff
@@ -0,0 +1,260 @@
+		self.assertTrue(os.path.exists(self.repo1.localdir))
+		dir_be_removed = self.repo1.localdir
+
+		self.assertEqual(self.repo1.remove(), None)  
+		self.assertEqual(self.repo1.localdir, None)
+		self.assertEqual(self.repo1.repo, None)
+		self.assertFalse(os.path.exists(dir_be_removed))
+		
+		# try to remove already removed repo
+		self.assertEqual(self.repo1.remove(), None)  
+
+	def test_clone(self):
+
+		self.assertRaises(TypeError, self.repo1.clone, 1)
+
+		self.assertEqual(self.repo1.clone(self.myname, self.myemail), None)  
+		localdir = self.repo1.localdir
+		self.assertTrue(os.path.exists(localdir))
+		self.assertIsNotNone(self.repo1.repo)
+		with self.repo1.repo.config_reader() as config:
+			self.assertEqual(config.get_value('user','email'), self.myemail)
+			self.assertEqual(config.get_value('user','name'), self.myname)
+		#print('Repo1 cloned into: ' + self.repo1.localdir)
+
+		self.assertEqual(self.repo2.clone(self.myname, self.myemail), None)  
+		localdir = self.repo2.localdir
+		self.assertTrue(os.path.exists(localdir))
+		self.assertIsNotNone(self.repo2.repo)
+		with self.repo2.repo.config_reader() as config:
+			self.assertEqual(config.get_value('user','email'), self.myemail)
+			self.assertEqual(config.get_value('user','name'), self.myname)
+		#print('Repo2 cloned into: ' + self.repo2.localdir)
+
+		self.assertEqual(self.repo3.clone(self.myname, self.myemail), None)  
+		localdir = self.repo3.localdir
+		self.assertTrue(os.path.exists(localdir))
+		self.assertIsNotNone(self.repo3.repo)
+		with self.repo3.repo.config_reader() as config:
+			self.assertEqual(config.get_value('user','email'), self.myemail)
+			self.assertEqual(config.get_value('user','name'), self.myname)
+		#print('Repo3 cloned into: ' + self.repo3.localdir)
+
+		with self.assertRaises(GitCommandError):
+			self.repo4.clone(self.myname, self.myemail)
+
+
+	def test_get_repo_dir(self):
+
+		self.assertEqual(self.repo1.clone(self.myname, self.myemail), None)  
+		self.assertEqual(self.repo2.clone(self.myname, self.myemail), None)  
+		self.assertEqual(self.repo3.clone(self.myname, self.myemail), None)  
+
+		repodir1 = self.repo1.get_repo_dir()
+		repodir2 = self.repo2.get_repo_dir()
+		repodir3 = self.repo3.get_repo_dir()
+
+		msg = 'git clone would strip the suffix .git while cloning the repo'
+		self.assertEqual(repodir1, get_repo_dir_without_git_suffix(self.repo1.repourl), msg)
+		self.assertEqual(repodir2, get_repo_dir_without_git_suffix(self.repo2.repourl), msg)
+		self.assertEqual(repodir3, get_repo_dir_without_git_suffix(self.repo3.repourl), msg)
+
+		repodir21 = self.repo1.localdir + '/' + repodir1
+		repodir22 = self.repo2.localdir + '/' + repodir2
+		repodir23 = self.repo3.localdir + '/' + repodir3
+
+		msg = 'Folder does not exist: ' 
+		self.assertTrue(os.path.exists(repodir21), msg + repodir21)
+		self.assertTrue(os.path.exists(repodir22), msg + repodir22)
+		self.assertTrue(os.path.exists(repodir23), msg + repodir23)
```

Comment:
you might receive folder does not exists in here. the 
`self.repo3.get_repo_dir()`
that you are using in lines 160-163 serve to different purpose. we are cloning the repo to directory with some kind of hash and not the name of the repository as you would do manually with git clone. The get_repo_dir serves to get that name if you want to use it to rename this hash or so.

To sum up this will not work in here because there will be no such folder if the folder inside of the repository isn t the same as the repo name